### PR TITLE
fix: use actual noteTitle in DeleteNoteSidebarEntry

### LINKED ISSUE #1728: Use actual note title in DeleteNoteSidebarEntry
**Which part of the project should be enhanced?**
DeleteNoteSidebarEntry

**Is your enhancement request related to a problem? Please describe.**
The DeleteNoteSidebarEntry does not show the title of the note, but a placeholder instead.

**Describe the solution you'd like**
Get the correct title from the redux state and display it.

**Describe alternatives you've considered**
Remove the mention of a note title completly.

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
DeleteNoteSidebarEntry

### Description
This PR fixes the DeleteNoteSidebarEntry component to use the actual note title and not just a placeholder string.

### Steps

- [x] Added implementation
- [x] Added / updated tests
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### Related Issue(s)
fixes #1728

### CODE REVIEW

File: src/components/editor-page/sidebar/delete-note-sidebar-entry.tsx [hunk 1]
```diff
@@ -4,32 +4,52 @@
...
         show={showDialog}
-        onHide={() => setShowDialog(false)}
+        onHide={closeDialog}
         title={'editor.modal.deleteNote.title'}>
         <h5>
           <Trans i18nKey={'editor.modal.deleteNote.question'} />
         </h5>
         <ul>
-          <li> noteTitle</li>
+          <li {...cypressId('deletion.noteTitle')}>&nbsp;{noteTitle}</li>
```
Comment (mrdrogdrog) [inline src/components/editor-page/sidebar/delete-note-sidebar-entry.tsx:52]: Why the nbsp?

File: src/components/editor-page/sidebar/delete-note-sidebar-entry.tsx [hunk 2]
```diff
@@ -4,32 +4,52 @@
...
 
   return (
     <Fragment>
-      <SidebarButton icon={'trash'} className={className} hide={hide} onClick={() => setShowDialog(true)}>
+      <SidebarButton
+        {...cypressId('menu-delete')}
+        icon={'trash'}
+        className={className}
+        hide={hide}
+        onClick={() => setShowDialog(true)}>
```
Comment (mrdrogdrog) [inline src/components/editor-page/sidebar/delete-note-sidebar-entry.tsx:38]: Extract callback

**Review by mrdrogdrog — CHANGES_REQUESTED** [review]

File: src/components/editor-page/sidebar/delete-note-sidebar-entry.tsx (hunk 1 above)
Comment (DerMolly) [reply src/components/editor-page/sidebar/delete-note-sidebar-entry.tsx:52]: Before it was a normal space, I think this is non-desirable therefore I replaced it with the nbsp

**Review by mrdrogdrog — APPROVED** [review]

File: src/components/editor-page/sidebar/delete-note-sidebar-entry/delete-note-sidebar-entry.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import { SidebarButton } from '../sidebar-button/sidebar-button'
+import type { SpecificSidebarEntryProps } from '../types'
+import { useApplicationState } from '../../../../hooks/common/use-application-state'
+import { cypressId } from '../../../../utils/cypress-attribute'
+import { showErrorNotification } from '../../../../redux/ui-notifications/methods'
+import { deleteNote } from '../../../../api/notes'
+import { DeleteNoteModal } from './delete-note-modal'
+import { ShowIf } from '../../../common/show-if/show-if'
+
+export const DeleteNoteSidebarEntry: React.FC<SpecificSidebarEntryProps> = ({ hide, className }) => {
```
Comment (ErikMichelson) [inline src/components/editor-page/sidebar/delete-note-sidebar-entry/delete-note-sidebar-entry.tsx:18]: ESDoc missing

**Review by ErikMichelson — CHANGES_REQUESTED** [review]

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**